onnx/defs/nn/defs.cc

```diff
@@ -1229,4 +1225,91 @@ ONNX_OPERATOR_SET_SCHEMA(
+            "The production MUST never overflow. The accumulation may overflow if and only if in 32 bits.")
+        .Input(
+            0,
+            "X",
+            "Input data tensor from previous layer; "
+            "has size (N x C x H x W), where N is the batch size, "
+            "C is the number of channels, and H and W are the "
+            "height and width. Note that this is for the 2D image. "
+            "Otherwise the size is (N x C x D1 x D2 ... x Dn)",
+            "T1")
+        .Input(
+            1,
+            "W",
+            "The weight tensor that will be used in the "
+            "convolutions; has size (M x C x kH x kW), where C "
+            "is the number of channels, and kH and kW are the "
+            "height and width of the kernel, and M is the number "
+            "of feature maps. For more than 2 dimensions, the "
+            "kernel shape will be (M x C x k1 x k2 x ... x kn), "
+            "where is the dimension of the kernel",
+            "T2")
+        .Input(
+            2,
+            "Z",
+            "padding value (zero_point normally), which should be a scalar tensor.",
+            "T1",
+            OpSchema::Optional)
+        .Output(
+            0,
+            "Y",
+            "Output data tensor that contains the result of the "
+            "convolution. The output dimensions are functions "
+            "of the kernel size, stride size, and pad lengths.",
+            "tensor(int32)")
+        .TypeConstraint(
+            "T1",
+            {"tensor(int8)",
+             "tensor(uint8)",
+             "tensor(int16)",
+             "tensor(uint16)"},
+            "Constrain input and output types to float tensors.")
+        .TypeConstraint(
+            "T2",
+            {"tensor(int8)",
+             "tensor(uint8)",
+             "tensor(int16)",
+             "tensor(uint16)"},
+            "Constrain input and output types to float tensors.")
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          // TODO: do we need to verify the input type to ensure both inputs (X
+          // and W) have same bit width?
+          // TODO: Should we enforce B has the same type as W?
+          convPoolShapeInference(ctx, true, false);
+        })
+        .Attr(
+            "kernel_shape",
+            "The shape of the convolution kernel. If not present, should be inferred from input W.",
+            AttributeProto::INTS,
+            OPTIONAL)
+        .Attr(
+            "dilations",
+            "dilation value along each axis of the filter. If not present, the dilation defaults to 1 along each axis.",
+            AttributeProto::INTS,
+            OPTIONAL)
+        .Attr(
+            "strides",
+            "Stride along each axis. If not present, the stride defaults to 1 along each axis.",
+            AttributeProto::INTS,
+            OPTIONAL)
+        .Attr(
```

Comment:
auto_pad is deprecated in the standard conv, so it shouldn't be there.

Reply:
ok. was thinking to remove it in separate PR for both conv and conv_integer later. will remove it right now.

Reply:
Think it twice, I'll still have the auto_pad removal in a separate PR for both conv and conv_integer soon.